src/compiler/tsc.ts

```diff
@@ -146,131 +173,176 @@ module ts {
             return sys.exit(EmitReturnStatus.CompilerOptionsErrors);
         }
 
-        if (compilerOptions.version) {
+        if (commandLine.options.version) {
             reportDiagnostic(createCompilerDiagnostic(Diagnostics.Version_0, version));
             return sys.exit(EmitReturnStatus.Succeeded);
         }
 
-        if (compilerOptions.help) {
+        if (commandLine.options.help) {
             printVersion();
             printHelp();
             return sys.exit(EmitReturnStatus.Succeeded);
         }
 
-        if (commandLine.filenames.length === 0) {
+        if (commandLine.options.project) {
+            configFilename = normalizePath(combinePaths(commandLine.options.project, "tsconfig.json"));

```

Comment:
i think an explicit message that the host does not support it, as opposed to the file is corrupt, no access privileges, or even does not exist, is more helpful. either ways i do not think we have a large user base on cscript anyways.